local/php/envs_test.go

```diff
@@ -182,6 +182,138 @@ func (s *PHPFPMSuite) TestGenerateEnv(c *C) {
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/",
+			expected: map[string]string{
+				"PATH_INFO":       "/",
+				"REQUEST_URI":     "/subdirectory/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/unknown.php",
+			expected: map[string]string{
+				"PATH_INFO":       "/unknown.php",
+				"REQUEST_URI":     "/subdirectory/unknown.php",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/unknown.php/",
+			expected: map[string]string{
+				"PATH_INFO":       "/unknown.php/",
+				"REQUEST_URI":     "/subdirectory/unknown.php/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/index.php/foo",
+			expected: map[string]string{
+				"PATH_INFO":       "/foo",
+				"REQUEST_URI":     "/subdirectory/index.php/foo",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory/subdirectory/",
+			expected: map[string]string{
+				"PATH_INFO":       "/",
+				"REQUEST_URI":     "/subdirectory/subdirectory/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "///subdirectory",
+			expected: map[string]string{
+				"PATH_INFO":       "///subdirectory",
+				"REQUEST_URI":     "///subdirectory",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/index.php",
+				"SCRIPT_NAME":     "/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/subdirectory///subdirectory//foo/",
+			expected: map[string]string{
+				"PATH_INFO":       "///subdirectory//foo/",
+				"REQUEST_URI":     "/subdirectory///subdirectory//foo/",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/subdirectory/index.php",
+				"SCRIPT_NAME":     "/subdirectory/index.php",
+			},
+		},
+		{
+			passthru: "/index.php",
+			uri:      "/../index.php",
+			expected: map[string]string{
+				"PATH_INFO":       "/../index.php",
+				"REQUEST_URI":     "/../index.php",
+				"QUERY_STRING":    "",
+				"SCRIPT_FILENAME": testdataDir + "/public/index.php",
+				"SCRIPT_NAME":     "/index.php",
+			},
```

Comment:
Yeah, i did mostly because i seen that curl doesn't resolve it, so although it's possible in browser it is possible in other environments. Said that, I am gonna drop bigger comment with how nginx deals with that in main topic.